modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -169,6 +186,64 @@ public void before() throws Exception
     }
 
     @Test
+    public void getMetadataUsesClassLoaderModelProperty() throws MetadataResolvingException
+    {
+        final MetadataResult result = mock(MetadataResult.class);
```

Comment:
I didn't get this. What I want to test is the following code in ExtensionComponent:
```
    @Override
    public MetadataResult<ComponentMetadataDescriptor> getMetadata() throws MetadataResolvingException
    {
        return withContextClassLoader(getClassLoader(this.extensionModel), () -> metadataMediator.getMetadata());
    }
```
When mediator is called the correct TCCL should be set. In order to test that I did use a custom Answer but at the mediator.getMetadata() mocked method. 

I didn't like it how the has to be written but didn't know another way.
Is there a better way to test this?